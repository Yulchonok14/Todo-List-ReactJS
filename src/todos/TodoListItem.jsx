import React from 'react';
import styled from 'styled-components';

const TodoItemContainer = styled.div`
    background: #fff;
    border-radius: 8px;
    margin-top: 8px;
    padding: 16px;
    position: relative;
    box-shadow: 0 4px 8px grey;
`;

export const getBorder = (startDate, today) => (
  startDate > new Date(today - 86400000 * 8) ? 'none' : '2px solid red'
);

const TodoItemContainerWithWarning = styled(TodoItemContainer)`
    border-bottom: ${(props) => getBorder(props.createdAt, Date.now())};
`;

const Button = styled.button`
    font-size: 16px;
    padding: 8px;
    border: none;
    border-radius: 8px;
    outline: none;
    cursor: pointer;
`;

const ButtonsContainer = styled.div`
    position: absolute;
    right: 12px;
    bottom: 12px;
`;

const RemoveButton = styled(Button)`
    display: inline-block;
    background-color: #ee2222;
    margin-left: 8px;
`;

const CompletedButton = styled(Button)`
    display: inline-block;
    background-color: #22ee22;
`;

const TodoListItem = ({ todo, onRemoveTodoClicked, onCompleteTodoClicked }) => {
  const Container = todo.isCompleted ? TodoItemContainer : TodoItemContainerWithWarning;
  return (
    <Container createdAt={todo.createdAt}>
      <h3>{todo.text}</h3>
      <p>
        Created at:&nbsp;
        {new Date(todo.createdAt).toLocaleDateString()}
      </p>
      <ButtonsContainer>
        {todo.isCompleted
          ? null
          : (
            <CompletedButton
              onClick={() => onCompleteTodoClicked(todo.id)}
            >
              Mark As Completed
            </CompletedButton>
          )}
        <RemoveButton
          onClick={() => onRemoveTodoClicked(todo.id)}
        >
          Remove
        </RemoveButton>
      </ButtonsContainer>
    </Container>
  );
};

export default TodoListItem;
