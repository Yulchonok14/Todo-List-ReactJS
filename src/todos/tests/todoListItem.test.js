/* eslint-disable react/jsx-filename-extension */
/* eslint-env mocha */
import chai, { expect } from 'chai';

import React from 'react';
import chaiJestSnapshot from 'chai-jest-snapshot';
import renderer from 'react-test-renderer';
import TodoListItem, { getBorder } from '../TodoListItem';

chai.use(chaiJestSnapshot);

before(() => {
  chaiJestSnapshot.resetSnapshotRegistry();
});

// eslint-disable-next-line func-names
beforeEach(function () {
  chaiJestSnapshot.configureUsingMochaContext(this);
});

describe('getBorder', () => {
  it('returns none border if created date is less than 8 days', () => {
    const createdDate = new Date(Date.now() - 86400000 * 3);
    const today = Date.now();
    const actual = getBorder(createdDate, today);
    const expected = 'none';

    expect(actual).be.equal(expected);
  });
  it('returns red border if created date is more than 8 days', () => {
    const createdDate = new Date(Date.now() - 86400000 * 10);
    const today = Date.now();
    const actual = getBorder(createdDate, today);
    const expected = '2px solid red';

    expect(actual).be.equal(expected);
  });
});

describe('TodoListItem', () => {
  const todoCompleted = [{
    id: '111',
    text: 'wash dishes',
    isCompleted: true,
    createdAt: Date.now(),
  }];

  it('renders correctly', () => {
    const component = renderer.create(<TodoListItem todo={todoCompleted} />);
    const tree = component.toJSON();

    expect(tree).to.matchSnapshot();
  });
});
