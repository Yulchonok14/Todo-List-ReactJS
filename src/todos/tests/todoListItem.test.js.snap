// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem renders correctly 1`] = `
<div
  className="sc-bdfBQB sc-gsTEea eiDaAo gLltEE"
>
  <h3 />
  <p>
    Created at:Â 
    Invalid Date
  </p>
  <div
    className="sc-hKgJUU eSVIjD"
  >
    <button
      className="sc-dlfnuX sc-jSgvzq gVyZRT iCLYda"
      onClick={[Function]}
    >
      Mark As Completed
    </button>
    <button
      className="sc-dlfnuX sc-eCstlR gVyZRT hwrRjV"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
</div>
`;
